geonode/maps/views.py

```diff
@@ -572,6 +581,15 @@ def map_metadata_detail(request, mapid, template="maps/map_metadata_detail.html"
             group = None
     site_url = settings.SITEURL.rstrip("/") if settings.SITEURL.startswith("http") else settings.SITEURL
     register_event(request, EventType.EVENT_VIEW_METADATA, map_obj)
+
+    # Load metadata_records for contrib apps
+    if getattr(settings, "EXTRA_METADATA_ENABLED", False):
```

Comment:
I have two thoughts about this:
1) I would evaluate having a function that puts in common this part of the code since it is repeated many times
2) is it actually needed to force the instance to have an additional field? the metadata itself as a json can be customized by changing the json structure in the settings:
https://github.com/GeoNode/geonode/blob/cb8d2aac6dae90ab61a266d902b5b1a38f957431/geonode/settings.py#L2161-L2167
It here looks to me like we are forcing the core to have a particular structure for an external library which creates a dependency. Instead by changing the structure of the settings you can "force" your own app to follow the schema that is declared. Saying this, the `extra_metadata` field and the `normal metadata field` can have the same structure shared between the core and the app and, in the end, is only the new template to define HTML as he wants (ps: the schema can change for each resource)